pkg/tsdb/prometheus/prometheus.go

```diff
@@ -22,6 +22,18 @@ type PrometheusExecutor struct {
 	Transport *http.Transport
 }
 
+type basicAuthTransport struct {
+	*http.Transport
+
+	username string
+	password string
+}
+
+func (bat basicAuthTransport) RoundTrip(req *http.Request) (*http.Response, error) {
+	req.SetBasicAuth(bat.username, bat.password)
+	return http.DefaultTransport.RoundTrip(req)
```

Comment:
I think the correct thing here would be to call `bat.Transport.Round()` rather then `http.DefaultTransport`.